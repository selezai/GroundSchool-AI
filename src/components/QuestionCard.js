import React from 'react';
import { View, Text, StyleSheet } from 'react-native';
import PropTypes from 'prop-types';

const QuestionCard = ({ 
  questionNumber,
  questionText,
  category,
  difficulty,
  testID = 'question-card'
}) => {
  return (
    <View style={styles.container} testID={testID}>
      <View style={styles.header}>
        <Text style={styles.questionNumber}>Question {questionNumber}</Text>
        <View style={styles.metadata}>
          <Text style={styles.category}>{category || 'General'}</Text>
          {difficulty ? (
            <Text style={[
              styles.difficulty,
              styles[difficulty.toLowerCase() || 'medium']
            ]}>
              {difficulty}
            </Text>
          ) : (
            <Text style={[styles.difficulty, styles.medium]}>Medium</Text>
          )}
        </View>
      </View>
      <Text style={styles.questionText}>{questionText}</Text>
    </View>
  );
};

QuestionCard.propTypes = {
  questionNumber: PropTypes.number.isRequired,
  questionText: PropTypes.string.isRequired,
  category: PropTypes.string.isRequired,
  difficulty: PropTypes.oneOf(['Easy', 'Medium', 'Hard']).isRequired,
  testID: PropTypes.string
};

const styles = StyleSheet.create({
  container: {
    backgroundColor: '#FFFFFF',
    borderRadius: 12,
    padding: 16,
    marginHorizontal: 16,
    marginVertical: 8,
    shadowColor: '#000',
    shadowOffset: { width: 0, height: 2 },
    shadowOpacity: 0.1,
    shadowRadius: 4,
    elevation: 3,
  },
  header: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    marginBottom: 12,
  },
  questionNumber: {
    fontSize: 16,
    fontWeight: '600',
    color: '#2D3748',
  },
  metadata: {
    flexDirection: 'row',
    alignItems: 'center',
  },
  category: {
    fontSize: 14,
    color: '#4A5568',
    marginRight: 8,
    backgroundColor: '#EDF2F7',
    paddingHorizontal: 8,
    paddingVertical: 4,
    borderRadius: 4,
  },
  difficulty: {
    fontSize: 14,
    fontWeight: '500',
    paddingHorizontal: 8,
    paddingVertical: 4,
    borderRadius: 4,
  },
  easy: {
    backgroundColor: '#C6F6D5',
    color: '#2F855A',
  },
  medium: {
    backgroundColor: '#FEEBC8',
    color: '#C05621',
  },
  hard: {
    backgroundColor: '#FED7D7',
    color: '#C53030',
  },
  questionText: {
    fontSize: 16,
    color: '#1A202C',
    lineHeight: 24,
  },
});

export default QuestionCard;
