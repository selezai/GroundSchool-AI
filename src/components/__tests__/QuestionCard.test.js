import React from 'react';
import renderer, { act } from 'react-test-renderer';
import QuestionCard from '../QuestionCard';

describe('QuestionCard Component', () => {
  const defaultProps = {
    questionNumber: 1,
    questionText: 'What is the minimum flight visibility for VFR flight in Class G airspace below 1,200 feet AGL during the day?',
    category: 'Regulations',
    difficulty: 'Medium'
  };

  it('renders correctly with default props', () => {
    let tree;
    act(() => {
      tree = renderer.create(<QuestionCard {...defaultProps} />).toJSON();
    });
    expect(tree).toMatchSnapshot();
  });

  it('renders with different difficulty levels', () => {
    const difficulties = ['Easy', 'Medium', 'Hard'];
    difficulties.forEach(difficulty => {
      let tree;
      act(() => {
        tree = renderer.create(
          <QuestionCard {...defaultProps} difficulty={difficulty} />
        ).toJSON();
      });
      expect(tree).toMatchSnapshot();
    });
  });

  it('renders with different categories', () => {
    const categories = ['Navigation', 'Weather', 'Aircraft Systems'];
    categories.forEach(category => {
      let tree;
      act(() => {
        tree = renderer.create(
          <QuestionCard {...defaultProps} category={category} />
        ).toJSON();
      });
      expect(tree).toMatchSnapshot();
    });
  });

  it('renders with long question text', () => {
    const longQuestion = 'A very long question that spans multiple lines and tests how the component handles text wrapping and layout adjustments when the content is significantly longer than usual. This helps ensure the component remains visually appealing with varying content lengths.';
    let tree;
    act(() => {
      tree = renderer.create(
        <QuestionCard {...defaultProps} questionText={longQuestion} />
      ).toJSON();
    });
    expect(tree).toMatchSnapshot();
  });
});
