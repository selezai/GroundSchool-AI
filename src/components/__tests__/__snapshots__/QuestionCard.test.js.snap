// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuestionCard Component renders correctly with default props 1`] = `null`;

exports[`QuestionCard Component renders with different categories 1`] = `null`;

exports[`QuestionCard Component renders with different categories 2`] = `null`;

exports[`QuestionCard Component renders with different categories 3`] = `null`;

exports[`QuestionCard Component renders with different difficulty levels 1`] = `null`;

exports[`QuestionCard Component renders with different difficulty levels 2`] = `null`;

exports[`QuestionCard Component renders with different difficulty levels 3`] = `null`;

exports[`QuestionCard Component renders with long question text 1`] = `null`;
